src/cargo/ops/tree/mod.rs

```diff
@@ -389,6 +422,11 @@ fn print_dependencies<'a>(
             // Filter out packages to prune.
             match graph.node(**dep) {
                 Node::Package { package_id, .. } => {
+                    if filter_non_workspace_member
+                        && !ws.is_member(graph.package_for_id(*package_id))
```

Comment:
nit: mind adding (in its own commit before this) a `is_member_id`?  We have to look up the package just for `Workspace` to use it to look up the id.  Seems like it'd be better if we called `is_member_id` here (and in other places)

Reply:
Sounds more convenient